.github/workflows/osx-tests.yml

```
@@ -0,0 +1,83 @@
+name: OSX Tests
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+
+jobs:
+  testing:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        setup: [unittests, functional, integration, performance]
```

```suggestion
        setup: [unittests, functional, integration]
```

performance tests are just meant for debugging but are not actually meant to be ran